packages/react-components/src/components/chart/baseChart.tsx

```diff
@@ -63,6 +67,15 @@ const Chart = ({ viewport, queries, size = { width: 500, height: 500 }, ...optio
 
   const { ref } = useECharts({
     option,
+    settings: {
+      /**
+       * if datastreams changes because we've modified the query
+       * ensure that series is updated - https://echarts.apache.org/en/api.html#echartsInstance.setOption
+       * TECHDEBT: hardcoding this prevents the normal merge animation from happening when new data points
+       * are added.
+       */
+      replaceMerge: ['series'],
```

Comment:
in the case of a series being added or deleted i.e query change, can we detect the delta and one update the options for one cycle / remove the series similar to the delete of the trend cursors ? replaceMerge may be a expensive and affect performance ?

Reply:
Ya, I had planned to circle back and optimize this some more. I left a techdebt tag so we don't forget. I want to include this in the refactor of the series / yAxis converter since changes to either should set this setting. I can do that now, but I was afraid it might cause too many conflicts?